config/configChecker.go

```diff
@@ -4,14 +4,47 @@ import (
 	"fmt"
 )
 
-// SanityCheckEnableEpochsStakingV4 checks if the enable epoch configs for stakingV4 are set correctly
-func SanityCheckEnableEpochsStakingV4(cfg *Configs) error {
-	enableEpochsCfg := cfg.EpochConfig.EnableEpochs
+// SanityCheckNodesConfig checks if the nodes limit setup is set correctly
+func SanityCheckNodesConfig(
+	nodesSetup NodesSetupHandler,
+	cfg EnableEpochs,
+) error {
+	maxNodesChange := cfg.MaxNodesChangeEnableEpoch
+	for _, maxNodesConfig := range maxNodesChange {
+		err := checkMaxNodesConfig(nodesSetup, maxNodesConfig)
+		if err != nil {
+			return fmt.Errorf("%w in MaxNodesChangeConfig at EpochEnable = %d", err, maxNodesConfig.EpochEnable)
+		}
+	}
+
+	return sanityCheckEnableEpochsStakingV4(cfg, nodesSetup.NumberOfShards())
+}
+
+func checkMaxNodesConfig(
+	nodesSetup NodesSetupHandler,
+	maxNodesConfig MaxNodesChangeConfig,
+) error {
+	nodesToShufflePerShard := maxNodesConfig.NodesToShufflePerShard
+	if nodesToShufflePerShard == 0 {
+		return errZeroNodesToShufflePerShard
```

Comment:
why this? Can't we have 0 nodes shuffling? Why should be this enforced?
For example: we might have a sovereign shard with 0 nodes shuffling as the owner of the shard does not want shuffling. Another example: chain simulator can be run with exactly 4 nodes and 4 keys.